platform/darwin/src/MGLFeature.mm

```diff
@@ -442,6 +443,10 @@ static CLLocationCoordinate2D toLocationCoordinate2D(const mbgl::Point<T> &point
     }
     shape.attributes = attributes;
 
+    if (MGLFeatureHasClusterAttribute(shape)) {
+        MGLConvertFeatureToClusterSubclass(shape);
```

Comment:
Consider modifying `GeometryEvaluator` to create an MGLCluster subclass instance directly:

https://github.com/mapbox/mapbox-gl-native/blob/0addd511d5fa503dc67d0773893c66c000e3ec98/platform/darwin/src/MGLFeature.mm#L320